pkg/controller/template/render.go

```diff
@@ -285,7 +285,7 @@ func generateMachineConfigForName(config *RenderConfig, role, name, templateDir,
 		return vs
 	}
 
-	ignCfg, err := ctrlcommon.TranspileCoreOSConfigToIgn(keySortVals(files), keySortVals(units))
+	ignCfg, err := ctrlcommon.ConvertButaneFragmentsToIgnition(keySortVals(files), keySortVals(units))
 	if err != nil {
 		return nil, fmt.Errorf("error transpiling CoreOS config to Ignition config: %v", err)
```

Comment:
Since you're in here, would you mind changing this error to:
```golang
fmt.Errorf("error transpiling CoreOS config to Ignition config: %w", err)
```

This will let us unwrap the original error later.